src/Orleans/Providers/ProviderTypeLoader.cs

```diff
@@ -87,6 +87,14 @@ private void ProcessAssemblyLocally(Assembly assembly)
 
         private static void ProcessNewAssembly(object sender, AssemblyLoadEventArgs args)
         {
+#if !NETSTANDARD
+            // If the assembly is loaded for reflection only avoid processing it.
+            if (args.LoadedAssembly.ReflectionOnly)
```

Comment:
I thought I had replied to this, but apparently not.

This was just me enumerating usages of `GetDefinedTypes` to see where it was being potentially used incorrectly. `AssemblyLoad` is called when asms are loaded `ReflectionOnly` - I tested in a clean project.